SPECS/python-tqdm/python-tqdm.spec

```diff
@@ -21,40 +21,83 @@ BuildArch:      noarch
 %{?python_provide:%python_provide python3-%{srcname}}
 Summary:        %{summary}
 BuildRequires:  python3-devel
+BuildRequires:  python3-pip
 BuildRequires:  python3-setuptools
 BuildRequires:  python3-setuptools_scm
 BuildRequires:  python3-wheel
+BuildRequires:  python3-packaging
+BuildRequires:  python3-setuptools_scm
+%if 0%{with_check}
+BuildRequires:  python3-pytest
+%ifarch x86_64
+BuildRequires:  python3-tensorflow
+%endif
+%endif
 
 %description -n python3-%{srcname} %{_description}
 
 Python 3 version.
 
 %prep
+export SETUPTOOLS_SCM_PRETEND_VERSION=%{version}
 %autosetup -n %{srcname}-%{version}
+chmod -x tqdm/completion.sh
+
+# https://github.com/tqdm/tqdm/pull/1292
+echo 'include tqdm/tqdm.1' >> MANIFEST.in
+echo 'include tqdm/completion.sh' >> MANIFEST.in
+
+%generate_buildrequires
+%pyproject_buildrequires -r
 
 %build
-%py3_build
+export SETUPTOOLS_SCM_PRETEND_VERSION=%{version}
+%pyproject_wheel
 
 %install
-%py3_install
+export SETUPTOOLS_SCM_PRETEND_VERSION=%{version}
+%pyproject_install
+%pyproject_save_files %{srcname}
 
-mkdir -p %{buildroot}%{_mandir}/man1/
-mv -v %{buildroot}%{python3_sitelib}/%{srcname}/%{srcname}.1 %{buildroot}%{_mandir}/man1/
+install -Dpm0644 \
+  %{buildroot}%{python3_sitelib}/tqdm/tqdm.1 \
+  %{buildroot}%{_mandir}/man1/tqdm.1
+install -Dpm0644 \
+  %{buildroot}%{python3_sitelib}/tqdm/completion.sh \
+  %{buildroot}%{_datadir}/bash-completion/completions/tqdm.bash
 
 %check
-pip3 install 'tox>=3.27.1,<4.0.0'
-tox -e setup.py
-
-%files -n python3-%{srcname}
+export SETUPTOOLS_SCM_PRETEND_VERSION=%{version}
+pip3 install --upgrade pip
+pip3 install iniconfig \
```

Comment:
I was wondering if we should pin the `pip installs` so that once in the future there is updates to the dependencies the tests do not break.

Reply:
Can you specify more about what you mean by pinning "pip installs" please? For this package, I don't think package test is ever enabled since python-pytest is never added as a build requirement and thus %pytest won't work.

Reply:
What I mean is e.g `pandas==x.y`